nca/NetworkConfig/ResourcesHandler.py

```diff
@@ -45,6 +57,37 @@ def set_global_peer_container(self, global_ns_list, global_pod_list, global_reso
         self._set_config_peer_container(global_ns_list, global_pod_list, global_resource_list,
                                         'global', True, global_resources_parser)
 
+    def analyze_livesim(self, peer_container, policy_finder):
+        """
+        Analyze the pre-parsing of the topology and finds what needs
+        to be added.
+        :param PeerContainer peer_container: Contains the peers found in pre-parsing
+        :param PolicyFinder policy_finder: Contains the policies found in pre-parsing
+        :return: [strings]: configuration_addons: the paths of the yamls to be added.
+        """
+        configuration_addons = []
+        path = os.path.dirname(__file__)
+        # find kube-dns reference
+        if policy_finder.missing_pods_with_labels.get('name') == 'kube-system':
+            configuration_addons.append(os.path.join(path, LiveSimPaths.DnsCfgPath))
+
+        # find ingress controller pods
+        if policy_finder.found_ingress_control_policy:
+            ingress_controller_found = False
+            ingress_controllers = ['ingress-nginx', 'ingress-gce', 'app-ingress']
+            for name in ingress_controllers:
+                if peer_container.get_pods_with_service_name_containing_given_string(name):
+                    ingress_controller_found = True
+                    break
```

Comment:
This is duplication of the check that is already done by the parser. Maybe instead of this duplication, the parser of ingress policy can have a member indicating if an ingress policy exists without an ingress controller pod, and this would be sufficient to add the livesim resource.

Reply:
done